src/averageAges.js

```diff
@@ -1,61 +1,100 @@
+    const centuryDied = Math.ceil(item.died / 100);
+
+    return century === centuryDied;
+  });
+
+  const sumAgeWithCentury = ageWithCentury.reduce((acc, current) => {
+    const age = current.died - current.born;
+
+    return acc + age;
+  }, 0);
+
+  const averageAge = sumaResult / arrayMen.length;
+  const averageAgeCentury = sumAgeWithCentury / ageWithCentury.length;
+
+  const result = century ? averageAgeCentury : averageAge;
+
+  return result;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const arrayWomens = people.filter(item => item.sex === 'f');
+
+  const ageAllWomens = arrayWomens.map(item => item.died - item.born);
+  const sumAgeAllWomens = ageAllWomens.reduce(
+    (acc, current) => acc + current, 0);
+  const averageAgeAllWomens = sumAgeAllWomens / arrayWomens.length;
+
+  const arrPeopleMother = people.map(item => item.mother);
+  const arrayWomensWithChild = arrayWomens.filter(
+    item => arrPeopleMother.includes(item.name));
+  const ageWomensWithChild = arrayWomensWithChild.map(
+    item => item.died - item.born);
+  const sumAgeWomenWithChild = ageWomensWithChild.reduce(
+    (acum, item) => acum + item, 0);
+  const averageAgeWomensWithChild = sumAgeWomenWithChild
+  / arrayWomensWithChild.length;
+
+  const result = withChildren ? averageAgeWomensWithChild : averageAgeAllWomens;
+
+  return result;
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrNamePeople = people.map(item => {
+    return item.name;
+  });
+
+  const arrPersWithMom = people.filter(item => {
+    return (item.mother !== null && arrNamePeople.includes(item.mother));
+  });
+
+  const arrAgeDiff = arrPersWithMom.map(item => {
+    const nameMother = item.mother;
+    const itemMother = people.filter(el => {
+      return el.name === nameMother;
+    });
+    const age = item.born - itemMother[0].born;
+
+    return age;
+  });
+
+  const sumAgeDiff = arrAgeDiff.reduce((accum, item) => accum + item, 0);
+  const averageAgeDiff = sumAgeDiff / arrAgeDiff.length;
+
+  const arrSonWithMom = arrPersWithMom.filter(item => item.sex === 'm');
```

Comment:
Instead of filtering the sons separately, you can include the condition for sons in the initial filter for people with mothers. This will simplify the code and reduce redundancy.